langchain4j-coherence/src/main/java/dev/langchain4j/store/embedding/coherence/CoherenceEmbeddingStore.java

```diff
@@ -0,0 +1,506 @@
+            throw new IllegalArgumentException("ids cannot be null or empty");
+        }
+
+        documentChunks.keySet().removeAll(ids.stream().map(DocumentChunk.Id::parse).collect(Collectors.toSet()));
+    }
+
+    @Override
+    public void removeAll(dev.langchain4j.store.embedding.filter.Filter filter) {
+        if (filter == null) {
+            throw new IllegalArgumentException("filter cannot be null");
+        }
+        
+        Filter<DocumentChunk> chunkFilter = CoherenceMetadataFilterMapper.map(filter);
+        documentChunks.invokeAll(chunkFilter, CacheProcessors.removeBlind());
+    }
+
+    @Override
+    public void removeAll() {
+        documentChunks.truncate();
+    }
+
+    @Override
+    public EmbeddingSearchResult<TextSegment> search(EmbeddingSearchRequest request) {
+        Embedding queryEmbedding = request.queryEmbedding();
+        if (normalizeEmbeddings) {
+            queryEmbedding.normalize();
+        }
+        double minScore = request.minScore();
+        boolean checkMin = minScore != 0.0f;
+        Filter<DocumentChunk> filter = CoherenceMetadataFilterMapper.map(request.filter());
+        Float32Vector vector = new Float32Vector(queryEmbedding.vector());
+
+        SimilaritySearch<DocumentChunk.Id, DocumentChunk, float[]> aggregator = new SimilaritySearch<>(EXTRACTOR, vector, request.maxResults());
+        if (filter != null) {
+            aggregator = aggregator.filter(filter);
+        }
+            
+        List<QueryResult<DocumentChunk.Id, DocumentChunk>> results = documentChunks.aggregate(aggregator);
+        List<EmbeddingMatch<TextSegment>> matches = new ArrayList<>();
+
+        for (QueryResult<DocumentChunk.Id, DocumentChunk> result : results) {
+            double score = RelevanceScore.fromCosineSimilarity(1.0f - result.getDistance());
+            if (checkMin && score < minScore) {
+                continue;
+            }
+
+            matches.add(embeddingMatch(score, result.getKey(), result.getValue()));
+        }
+
+        matches.sort(COMPARATOR);
+        return new EmbeddingSearchResult<>(matches.size() > request.maxResults()
+                                           ? matches.subList(0, request.maxResults())
+                                           : matches);
+    }
+
+    /**
+     * Perform a KNN search.
+     *
+     * @param vector  the vector to find the nearest neighbours to
+     * @param k       the maximum number of neighbours to find
+     *
+     * @return  the search results
+     */
+    public List<QueryResult<DocumentChunk.Id, DocumentChunk>> search(float[] vector, int k) {
+         return documentChunks.aggregate(new SimilaritySearch<>(DocumentChunk::vector, new Float32Vector(vector), k));
+     }
+
+    /**
+     * Obtain the {@link NamedMap} used to store embeddings.
+     *
+     * @return the {@link NamedMap} used to store embeddings
+     */
+    public NamedMap<DocumentChunk.Id, DocumentChunk> getDocumentChunks() {
```

Comment:
```suggestion
    public NamedMap<DocumentChunk.Id, DocumentChunk> documentChunks() {
```

Reply:
This is not needed either. Removed altogether.